crates/electrum/src/electrum_ext.rs

```diff
@@ -134,64 +134,54 @@ pub struct ElectrumUpdate {
 
 /// Trait to extend [`Client`] functionality.
 pub trait ElectrumExt {
-    /// Scan the blockchain (via electrum) for the data specified and returns updates for
-    /// [`bdk_chain`] data structures.
+    /// Full scan the keychain scripts specified with the blockchain (via an Electrum client) and
+    /// returns updates for [`bdk_chain`] data structures.
     ///
     /// - `prev_tip`: the most recent blockchain tip present locally
     /// - `keychain_spks`: keychains that we want to scan transactions for
-    /// - `txids`: transactions for which we want updated [`Anchor`]s
-    /// - `outpoints`: transactions associated with these outpoints (residing, spending) that we
-    ///     want to included in the update
     ///
-    /// The scan for each keychain stops after a gap of `stop_gap` script pubkeys with no associated
+    /// The full scan for each keychain stops after a gap of `stop_gap` script pubkeys with no associated
     /// transactions. `batch_size` specifies the max number of script pubkeys to request for in a
     /// single batch request.
-    fn scan<K: Ord + Clone>(
+    fn full_scan<K: Ord + Clone>(
         &self,
-        prev_tip: CheckPoint,
+        prev_tip: &CheckPoint,
```

Comment:
I think we shouldn't use references here. Although we can it seems like the wrong hint to give to the user. The design of all the other parameters is that this function owns all of them so there can be no lifetime issues when spawning off a new thread to run this. The value returned by `LocalChain::tip` should be passed trait into this -- borrowing it beforehand doesn't help anyone and just leads to a compiler error before you add a reference and pass that it instead.

Reply:
Fixed.